pandas/tests/indexes/interval/test_interval.py

```diff
@@ -435,9 +438,12 @@ def test_maybe_convert_i8_numeric(self, breaks, make_key):
         index = IntervalIndex.from_breaks(breaks)
         key = make_key(breaks)
 
-        # no conversion occurs for numeric
         result = index._maybe_convert_i8(key)
-        assert result is key
+        if not isinstance(result, NumericIndex):
+            assert result is key
+        else:
+            expected = NumericIndex(key)
+            tm.assert_index_equal(result, expected)
```

Comment:
rather than adding logic to the test (which can hide bugs), is it possible to either:
- include the expected result in the parametrisation
- OR split the test out into two separate ones, one of which uses `assert result is key` and the other `tm.assert_index_equal(result, expected)`?

Reply:
Also, doesn't this test also pass on `upstream/main`? Is there a way to write it such that it fails there, but passes here?

Reply:
I didn't get time to address this question right now, sorry. I'll get back to this tonight.

Reply:
I`ve updated the PR.

Reply:
thanks for updating - any chance you could address the logic in the test comment too please?

IIRC when I tried executing this, it was just one of the `make_key` inputs which required a different assertion

If so, then can the assertion either be included in the parametrisation, or the test be split into two?

For reference, this advice comes from: https://testing.googleblog.com/2014/07/testing-on-toilet-dont-put-logic-in.html

Reply:
Yeah good points int the article, very nice to have it articulated

I've changed the PR. I started by separating the test into two tests split by type of make key. However, I didn't like having two very similar tests and I didn't like having lambdas in parametrization, (because lambdas are difficult to introspect, and having several lambdas means it's difficult to see which test you're looking at when debugging). so I've made a new version.

I prefer the newest version (avoiding lambdas, clear inputs into the test function), but will await your comment.

Reply:
sure, looks better, thanks for updating